src/components/TimeseriesDataExport/TimeseriesDataExport.tsx

```diff
@@ -118,17 +95,35 @@ const TimeseriesDataExportFC: FC<TimeseriesDataExportFormProps> = (
     imageDownloadBtn,
   } = lang;
 
+  const fetchDataPoints = async (request: DatapointsMultiQuery) => {
+    const { start = 0, end = 0 } = request;
+    const numericGranularity = getGranularityInMS(granularity);
+    const msPerRequest = numericGranularity * CELL_LIMIT / timeseriesIds.length;
+
+    const limits = range(+start, +end, msPerRequest);
+    const lastLimit = limits.length % 2 === 0 ? [+end - msPerRequest, +end] : [+end];
+    const ranges = chunk([...limits, ...lastLimit], 2);
+    const limit = CELL_LIMIT / timeseriesIds.length;
+
+    const requests = ranges
+      .map(([start, end]) => ({ ...request, start, end, limit }))
+      .map(request => context!.datapoints.retrieve(request));
+
+    return (await Promise.all(requests)).flat();
```

Comment:
While this could possibly work it also might provide slightly invalid data at some point.
The thing is that we need to have API as a single source of truth. By calculating start and end values ourself we might miss some points due to different way of rounding up in frontend vs backend.

The safe way is to iterate on api calls, it is explained by timelords here https://cognitedata.atlassian.net/wiki/spaces/TTSD/pages/1113227496/Paginating+datapoints
Quoting it here as well:

```
Case 2: Retrieving aggregates, includeOutsidePoints = false 
For aggregate retrieval, use a limit of 10,000.

Decide on how non-aligned start/stop should be aligned with the granularity (the API rounds start down, the PySDK rounds up before requesting)
Retrieve the a page of aggregates with the current start/stop. 
If the returned number of datapoints is less than 10k, stop
Otherwise, set the start to (timestamp of last returned datapoint) + (granularity in ms), and continue retrieval
```

Reply:
This was the source of troubles for my initial version of this in the SDK as well. The problem with sequential iteration is it quickly gets very inefficient for larger queries. 500k datapoints is suddenly 50 requests and altough you should expect some waiting, doing those sequentially will result in almost a minute of waiting even on a good network connection. 

Firing the requests independently and stiching them together is potentially _a lot_ faster.

Reply:
First of all, thank you for your thorough review. I will also add unit-tests and update storybook of course.

Regarding the fetching part, I would really like to avoid serial-fetching. I would even prefer always sending one request more than needed, and shaving of unneeded points afterwards.

Reply:
@hernil it won't be 50x faster due to browser limit, but yeah, you can expect 4-5 times better results.

@sanderland could you give a hint on how to clean-up results using this approach? (I know you worked a bit more with it)

Reply:
yes, please contact me on slack for a video chat or something and I can explain how paralell datapoint fetching is done in at least 2 ways.

Reply:
When is a good time for a video chat @sanderland ?

Reply:
tomorrow 10:30 am?

Reply:
That might crash with our standup, are you available some time after 1130am?

Reply:
3pm?

Reply:
slack is much better than this comment thread btw

Reply:
Perfect!